### PR TITLE
Support Python 3.9-3.13

### PR DESCRIPTION
**Pull request by thisac**
* Add support for Python 3.13
* Deprecate support for Python 3.8

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -210,7 +210,7 @@ workflows:
       - build-and-test-linux: &build
           matrix:
             parameters:
-              python-version: &python-versions [3.8.9, 3.9.4, 3.10.0, 3.11.0, 3.12.0]
+              python-version: &python-versions [3.9.4, 3.10.0, 3.11.0, 3.12.0, 3.13.0rc3]
```
Comment (randomir) [inline .circleci/config.yml]: Can we go to 3.13, or still waiting cibuildwheel?

Comment (arcondello) [reply .circleci/config.yml]: They just now merged 3.13 PRs, so imagine release is imminent. https://github.com/pypa/cibuildwheel/pull/2035, https://github.com/pypa/cibuildwheel/pull/2032

Comment (arcondello) [reply .circleci/config.yml]: cibuildwheel [has been released](https://github.com/pypa/cibuildwheel/releases/tag/v2.21.3)

Comment (thisac) [reply .circleci/config.yml]: Updated to use 3.13 now. Waiting for review. @arcondello @randomir

**Review by arcondello — APPROVED** [review]

**Review by randomir — APPROVED** [review]

**Pull request merged**